### ISSUE TITLE
task alternative: https://github.com/upcmd/up 

### ISSUE THREAD

**Issue by stephencheng** [go-task/task#348]
Sorry to pop around here

I have come up an alternative ground up implementation following up the original issue raised myself sometime ago: https://github.com/go-task/task/issues/208 , repo: https://github.com/upcmd/up

The project is to some extend inspired by go-task and I'd alway respect this origin

What I will really appreciate is that if the auth(s) could have a quick look and see if you have some more valuable feedback. I see the open source is to really make nice things to the community and I see this is opportunity than competition.

My main concern is that with all the features I put there, I convinced me not to do more but less, for example concurrency, file watcher etc. I feel the valuable input from experienced devops and someone in this domain would be very valuable for the future of the project

Thanks

**Comment by andreynering** [comment]
Hi @stephencheng,

Nice to see yet another Task-inspired tool emerging.

Unfortunately I do not have the time to check the tool in depth.

I think that one suggestion I have is: invest on documentation. I think that one of the most important things that helped Task attract user is having a clear and succinct documentation.

**Comment by stephencheng** [comment]
Thanks @andreynering 

I totally agree the documentation is very important. Without clear examples, it's hard for user to understand and use it at all. 

I found go-task's documentation site is really clear and clean. And actually I have gone through all examples cases thoroughly. 

I have put the document site online firstly before the formal release of the project: https://upcmd.netlify.app/

How do you ensure the data consistency/thread safe if go-task support run tasks in parallel? I found this is tricky, not only it generates lots of effort to implement it in a safe way, but also it's hard for user to use locking mechanism and then result in complication on top of using the tool. Just curious how this is implemented.

Thanks
